src/main/java/com/amazon/opendistroforelasticsearch/security/rest/TenantInfoAction.java

```diff
@@ -137,7 +144,37 @@ public void accept(RestChannel channel) throws Exception {
             }
         };
     }
-    
+
+    private boolean isUnAuthorized() {
+        final User user = (User)threadContext.getTransient(ConfigConstants.OPENDISTRO_SECURITY_USER);
+
+        // check if the user is a kibanauser or super admin
+        boolean initialCheck =   user == null ||
+                (!user.getName().equals(evaluator.kibanaServerUsername()))
+                        && !adminDns.isAdmin(user);
+        if(!initialCheck){
+            return false;
+        }
+
+        // check if the roles belong to kibana opendistro role
+        final SecurityDynamicConfiguration<?> configuration = load(CType.ROLESMAPPING, true);
```

Comment:
not sure if this operation is expensive, but you can move it inside the "if" section where you check if "kibanaOpendistroRole" is present

Reply:
checking kibanaOpendistroRole itself require configuration.
`if (configuration.exists(kibanaOpendistroRole)) {`
both configuration and kibanaOpendistroRole initialization should be before the check

Reply:
correct me if I am missing something here